### PR TITLE
Fix dataset import for Datumaro format

### LINKED ISSUE #4410: Export and subsequent import causes Error 500 - Datumaro format
### My actions before raising this issue
- [x] Read/searched [the docs](https://github.com/opencv/cvat/tree/master#documentation)
- [x] Searched [past issues](/issues)

### Expected Behaviour

Export and subsequent import of the same zip causes a **Error 500**

### Current Behaviour

I exported a dataset in Datumaro format. For testing the workflow, I created a new project, imported the exported folder but I received 
 `Error: Request failed with status code 500. "TypeError: _import() takes 2 positional arguments but 3 were given\n".`

### Steps to Reproduce (for bugs)

1. Export a dataset in Datumaro format
1. Import it into the platform

### Your Environment
Online platform - https://cvat.org/

### Next steps
You may [join our Gitter](https://gitter.im/opencv-cvat/public) channel for community support.

### PR DESCRIPTION
**Pull request by kirill-sizov**
### Motivation and context

Fix #4410 

- Fixed project import and export for Datumaro, KITTI and VGGFace2 formats
- Added project import/export tests

### Checklist

- [x] I submit my changes into the `develop` branch
- [x] I have added a description of my changes into [CHANGELOG](https://github.com/opencv/cvat/blob/develop/CHANGELOG.md) file
- [x] I have updated the [documentation](
  https://github.com/opencv/cvat/blob/develop/README.md#documentation) accordingly
- [x] I have added tests to cover my changes
- [x] I have linked related issues ([read github docs](
  https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword))
- [x] I have increased versions of npm packages if it is necessary ([cvat-canvas](https://github.com/opencv/cvat/tree/develop/cvat-canvas#versioning),
[cvat-core](https://github.com/opencv/cvat/tree/develop/cvat-core#versioning), [cvat-data](https://github.com/opencv/cvat/tree/develop/cvat-data#versioning) and [cvat-ui](https://github.com/opencv/cvat/tree/develop/cvat-ui#versioning))

### License

- [x] I submit _my code changes_ under the same [MIT License](
  https://github.com/opencv/cvat/blob/develop/LICENSE) that covers the project.
  Feel free to contact the maintainers if that's a concern.
- [x] I have updated the license header for each file (see an example below)

```python
# Copyright (C) 2022 Intel Corporation
#
# SPDX-License-Identifier: MIT
```

### CODE REVIEW

File: cvat/apps/dataset_manager/bindings.py [hunk 1]
```diff
@@ -1643,8 +1643,8 @@ def load_dataset_data(project_annotation, dataset: Dataset, project_data):
                 dataset_files['media'] += \
                     list(map(lambda ri: ri.path, dataset_item.related_images))
 
-        shortes_path = min(dataset_files['media'], key=lambda x: len(Path(x).parts), default=None)
-        if shortes_path is not None:
-            dataset_files['data_root'] = str(Path(shortes_path).parent.absolute()) + osp.sep
+        shortes_path = osp.commonpath(dataset_files['media'])
+        if shortes_path:
+            dataset_files['data_root'] = osp.abspath(shortes_path) + osp.sep
```
Comment (zhiltsov-max) [inline cvat/apps/dataset_manager/bindings.py:1648]: The same comments as in #4861

Comment (yasakova-anastasia) [reply cvat/apps/dataset_manager/bindings.py]: Fixed

File: tests/python/rest_api/test_projects.py [hunk 2]
```diff
@@ -458,6 +458,23 @@ def test_admin_can_get_project_backup_and_create_project_by_backup(self, admin_u
...
+
+        tmp_file = io.BytesIO(response.data)
+        tmp_file.name = "dataset.zip"
+
+        import_data = {
+            "dataset_file": tmp_file,
+        }
+
+        self._test_import_project(username, project_id, format_name, import_data)
+
```
Comment (zhiltsov-max) [inline tests/python/rest_api/test_projects.py:477]: I don't think it makes sense to test this twice. It should already be tested in `cvat/apps/dataset_manager/tests/test_rest_api_formats.py`, which is also modified in this PR.

**Comment by yasakova-anastasia** [comment]
> @yasakova-anastasia , could you also update the formats section of README.md?

Updated.

File: tests/python/rest_api/test_projects.py (hunk 2 above)
Comment (yasakova-anastasia) [reply tests/python/rest_api/test_projects.py:495]: I think this test is needed. It refers to a specific issue. The test as here is the most understandable and transparent.

File: README.md [hunk 3]
```diff
@@ -150,7 +151,9 @@ For more information about the supported formats, look at the
 | [Open Images V6](https://storage.googleapis.com/openimages/web/index.html)                                |   ✔️    |   ✔️    |
 | [Cityscapes](https://www.cityscapes-dataset.com/login/)                                                   |   ✔️    |   ✔️    |
 | [KITTI](http://www.cvlibs.net/datasets/kitti/)                                                            |   ✔️    |   ✔️    |
+| [Kitti Raw Format](https://www.cvlibs.net/datasets/kitti/raw_data.php)                                    |   ✔️    |   ✔️    |
 | [LFW](http://vis-www.cs.umass.edu/lfw/)                                                                   |   ✔️    |   ✔️    |
+| [Sly Point Cloud Format](https://docs.supervise.ly/data-organization/00_ann_format_navi)                  |   ✔️    |   ✔️    |
```
Comment (zhiltsov-max) [inline README.md:156]:
```suggestion
| [Supervisely Point Cloud Format](https://docs.supervise.ly/data-organization/00_ann_format_navi)                  |   ✔️    |   ✔️    |
```

File: cvat/apps/dataset_manager/tests/test_rest_api_formats.py [hunk 4]
```diff
@@ -1228,41 +1236,79 @@ def test_api_v2_export_dataset(self):
...
+                        if upload_format_name in tasks:
+                            project['labels'] = tasks[upload_format_name]['labels']
+                        project = self._create_project(project)
+                        file_zip_name = osp.join(test_dir, f"{test_name}_{edata['name']}_{upload_format_name}.zip")
+                        url = self._generate_url_upload_project_dataset(project['id'], upload_format_name)
+
+                        if osp.exists(file_zip_name):
+                            with open(file_zip_name, 'rb') as binary_file:
+                                response = self._post_request_with_data(url, {"dataset_file": binary_file}, user)
+                                print(response.status_code)
```
Comment (SpecLad) [inline cvat/apps/dataset_manager/tests/test_rest_api_formats.py:1308]: I think you forgot to remove debug output.

File: CHANGELOG.md [hunk 5]
```diff
@@ -138,6 +138,7 @@ Skeleton (<https://github.com/cvat-ai/cvat/pull/1>), (<https://github.com/opencv
 - Fix build dependencies for Siammask (<https://github.com/openvinotoolkit/cvat/pull/4486>)
 - Bug: Exif orientation information handled incorrectly (<https://github.com/openvinotoolkit/cvat/pull/4529>)
 - Fixed build of retinanet function image (<https://github.com/cvat-ai/cvat/pull/54>)
+- Dataset import for Datumaro format (<https://github.com/opencv/cvat/pull/4544>)
```
Comment (SpecLad) [inline CHANGELOG.md:141]: If this is about the Datumaro format, why are there also changes to other formats (KITTI and VGGFace)?

File: cvat/apps/dataset_manager/tests/test_rest_api_formats.py [hunk 6]
```diff
@@ -1228,41 +1236,79 @@ def test_api_v2_export_dataset(self):
...
+                    task = self._create_task(tasks['task in project #1'], self._generate_task_images(3))
+
                     url = self._generate_url_dump_project_dataset(project['id'], dump_format_name)
 
+                    if dump_format_name in [
+                        "MOT 1.1", "MOTS PNG 1.0", \
+                        "PASCAL VOC 1.1", "Segmentation mask 1.1", \
+                        "TFRecord 1.0", "YOLO 1.1", "ImageNet 1.0", \
+                        "WiderFace 1.0", "VGGFace2 1.0", "Cityscapes 1.0", \
+                        "Datumaro 1.0"\
```
[the remark this answers is not in the document]
Comment (yasakova-anastasia) [reply cvat/apps/dataset_manager/tests/test_rest_api_formats.py]: Updated

File: CHANGELOG.md (hunk 5 above)
Comment (yasakova-anastasia) [reply CHANGELOG.md]: The issue is related to the Datumaro format. Updated Changelog.

File: cvat/apps/dataset_manager/tests/test_rest_api_formats.py [hunk 7]
```diff
@@ -1229,40 +1237,82 @@ def test_api_v2_export_dataset(self):
...
-                    tasks['task in project #1']['project_id'] = pid
-                    self._create_task(tasks['task in project #1'], images)
-                    images = self._generate_task_images(3, 3)
-                    tasks['task in project #2']['project_id'] = pid
-                    self._create_task(tasks['task in project #2'], images)
+                    if dump_format_name in [
+                        'Cityscapes 1.0', 'LFW 1.0', 'Market-1501 1.0',
+                        'MOT 1.1', 'TFRecord 1.0'
+                    ]:
+                        self.skipTest("TO-DO: fix bug for this formats")
```
Comment (SpecLad) [inline cvat/apps/dataset_manager/tests/test_rest_api_formats.py:1244]: `skipTest` and `subTest` do not play well together. I did some testing, and when `pytest` is used as the test runner, `skipTest` just skips the entire rest of the test (when `unittest` is used as the runner, only the subtest is skipped, but the whole test is _marked_ skipped in the log, which is not great either).

**Review by SpecLad — APPROVED** [review]
LGTM, although the GitHub actions didn't run for some reason? Might need to push the branch again.

File: tests/python/rest_api/test_projects.py (hunk 2 above)
Comment (zhiltsov-max) [reply tests/python/rest_api/test_projects.py:495]: As I see, this test is also a complete duplicate of the test above. Maybe we can at least merge their codes and add a parameter?

**Comment by bsekachev** [comment]
We have different test failed in two runnings..
- actions_tasks3/case_19_all_image_rotate_features.js 
- actions_tasks3/case_18_filters_functionality.js

**Pull request merged**